src/Razor/Razor.Runtime/src/Runtime/TagHelpers/TagHelperRunner.cs

```diff
@@ -28,24 +29,23 @@ public Task RunAsync(TagHelperExecutionContext executionContext)
             }
 
             var tagHelperContext = executionContext.Context;
-            var tagHelpers = executionContext.TagHelpers;
-            OrderTagHelpers(tagHelpers);
+            var tagHelpers = CollectionsMarshal.AsSpan(executionContext.TagHelperList);
 
-            // Read interface .Count once rather than per iteration
-            var count = tagHelpers.Count;
-            for (var i = 0; i < count; i++)
+            tagHelpers.Sort(default(SortTagHelpers));
```

Comment:
Out of curiosity, is there a reason this is preferable to calling `executionContext.TagHelperList.Sort(default(SortTagHelpers))`?

Reply:
List Sort will create a Span and then use Span Sort; however we want the Span anyway; so might as well get that first and then sort the Span?

Iterating the Span is way faster than iterating the List; as it doesn't bounds check (or version check for `foreach`) each iteration etc, which is mostly what this is about.